californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/ObserveServerSideTest.java

```diff
@@ -403,31 +404,30 @@ public void testQuickChangeAndTimeout() throws Exception {
 
 		// First notification
 		testObsResource.change("First notification " + generateRandomPayload(10));
-		client.expectResponse().type(NON).code(CONTENT).token(tok).storeMID("MID").checkObs("A", "B").payload(respPayload).go();
+		client.expectResponse().type(NON).code(CONTENT).token(tok).newMID("MID").checkObs("A", "B").payload(respPayload).go();
 
 		// Now client crashes and no longer responds
 
 		respType = CON;
 		testObsResource.change("Second notification " + generateRandomPayload(10));
-		client.expectResponse().type(respType).code(CONTENT).token(tok).checkObs("B", "C").payload(respPayload).go();
+		client.expectResponse().type(respType).code(CONTENT).token(tok).newMID("MID").checkObs("B", "C").payload(respPayload).go();
 		// client does not ACK the CON notification
 
 		respType = NON;
 		testObsResource.change("NON notification 1 " + generateRandomPayload(10));
 		// server re-transmits unACKed CON notification but client does not reply
-		client.expectResponse().type(CON).code(CONTENT).token(tok).checkObs("B", "B").payload(respPayload).go();
+		client.expectResponse().type(CON).code(CONTENT).token(tok).newMID("MID").checkObs("B", "B").payload(respPayload).go();
```

Comment:
If the payload is changed, a new MID is required!

That's the above mentioned issue #258 

See 

https://github.com/eclipse/californium/issues/258#issuecomment-292884306